ts-client/README.md

```diff
@@ -1,7 +1,180 @@
+const provider = new AnchorProvider(mainnetConnection, mockWallet, {
+  commitment: 'confirmed',
+});
+// Alternatively, to use Solana Wallet Adapter
+
+const constantProductPool = await AmmImpl.create(connection, MAIN_POOL_USDT_SOL);
+const stablePool = await AmmImpl.create(connection, MAIN_POOL_USDT_USDC);
+```
+
+3. To interact with the AmmImpl
+
+- Get Lp Supply
+
+```ts
+// To refetch the pool's latest supply
+// Alternatively, use `AmmImpl.poolState.lpSupply`
+const lpSupply = await constantProductPool.getLpSupply();
+```
+
+- Check pool balance
+
+```ts
+// Get the user's ATA LP balance
+const userLpBalance = await constantProductPool.getUserBalance(mockWallet.publicKey);
+```
+
+- Deposit to constant product
+
+```ts
+const inAmountALamport = new BN(1 * 10 ** constantProductPool.tokenA.decimals); // 1.0 SOL
+
+// Get deposit quote for constant product
+const { poolTokenAmountOut, tokenAInAmount, tokenBInAmount } = await constantProductPool.getDepositQuote(
+  inAmountALamport,
+  new BN(0),
+);
+
+const depositTx = await constantProductPool.deposit(
+  mockWallet.publicKey,
+  tokenAInAmount,
+  tokenBInAmount,
+  poolTokenAmountOut,
+); // Web3 Transaction Object
+const depositResult = await provider.sendAndConfirm(depositTx); // Transaction hash
+```
+
+- Balance deposit to stable pool
+
+```ts
+const inAmountALamport = new BN(0.1 * 10 ** stablePool.tokenA.decimals);
+
+const { poolTokenAmountOut, tokenAInAmount, tokenBInAmount } = await stablePool.getDepositQuote(
+  inAmountALamport,
+  new BN(0),
+);
+
+const depositTx = await stablePool.deposit(mockWallet.publicKey, tokenAInAmount, tokenBInAmount, poolTokenAmountOut); // Web3 Transaction Object
+const depositResult = await provider.sendAndConfirm(depositTx); // Transaction hash
+```
+
+- Double side imbalance deposit to stable pool
+
+```ts
+const inAmountALamport = new BN(0.1 * 10 ** stablePool.tokenA.decimals);
+const inAmountBLamport = new BN(0.1 * 10 ** stablePool.tokenB.decimals);
+
+const { poolTokenAmountOut, tokenAInAmount, tokenBInAmount } = await stablePool.getDepositQuote(
+  inAmountALamport,
+  inAmountBLamport,
+  false, // pass in false for imbalance deposit quote
+); // Web3 Transaction Object
+const depositTx = await stablePool.deposit(mockWallet.publicKey, tokenAInAmount, tokenBInAmount, poolTokenAmountOut);
+const depositResult = await provider.sendAndConfirm(depositTx); // Transaction hash
+```
+
+- Single side imbalance deposit to stable pool
+
+```ts
+const inAmountALamport = new BN(0.1 * 10 ** stablePool.tokenA.decimals);
+
+const { poolTokenAmountOut, tokenAInAmount, tokenBInAmount } = await stablePool.getDepositQuote(
+  inAmountALamport,
+  new BN(0),
+  false, // pass in false for imbalance deposit quote
+); // Web3 Transaction Object
+const depositTx = await stablePool.deposit(mockWallet.publicKey, tokenAInAmount, tokenBInAmount, poolTokenAmountOut);
+const depositResult = await provider.sendAndConfirm(depositTx); // Transaction hash
+```
```

Comment:
1. Can u explain what's wrong ?
2. There are a few way integrators can use imbalance (single side & double side), i think we should show all the way

Reply:
I think it generally is very hard to read, and to sum up what is happening
Let's combine some of the permutation to make it easily glanceable 

1. combine `Deposit to constant product` and `Balance deposit to stable pool`
- Just write a minor description that says it works the same for both CP and stable

2. combine `Double side imbalance deposit to stable pool` and `Single side imbalance deposit to stable pool`
- since it's `imbalanced` it should be self-explanatory
```ts
const inAmountALamport = new BN(0.1 * 10 ** stablePool.tokenA.decimals);

const { poolTokenAmountOut, tokenAInAmount, tokenBInAmount } = stablePool.getDepositQuote(
  inAmountALamport, // pass in any amount, or even zero (new BN(0))
  new BN(0), // pass in any amount, or even zero (new BN(0))
  false, // pass in false for imbalance deposit quote
); // Web3 Transaction Object
const depositTx = await stablePool.deposit(mockWallet.publicKey, tokenAInAmount, tokenBInAmount, poolTokenAmountOut);
const depositResult = await provider.sendAndConfirm(depositTx); // Transaction hash
```

Reply:
3. combine these 3 cases, `Balance withdraw from constant product`, `Balance withdraw from stable pool`, `Imbalance withdraw from stable pool`

- just show them the second parameter is for imbalanced withdraw
- and add a minor description that says it works the same for both CP and stable


```ts
const outTokenAmountLamport = new BN(0.1 * 10 ** stablePool.decimals);

// Balanced withdraw
const { poolTokenAmountIn, tokenAOutAmount, tokenBOutAmount } = stablePool.getWithdrawQuote(
  outTokenAmountLamport,
);

// Imbalanced withdraw
const { poolTokenAmountIn, tokenAOutAmount, tokenBOutAmount } = stablePool.getWithdrawQuote(
  outTokenAmountLamport,
  new PublicKey(stablePool.tokenA.address), // Pass in your desired A/B token for imbalanced withdraw
);


const withdrawTx = await stablePool.withdraw(mockWallet.publicKey, poolTokenAmountIn, tokenAOutAmount, tokenBOutAmount); // Web3 Transaction Object
const withdrawResult = await provider.sendAndConfirm(withdrawTx);
```

Reply:
so we only need 2 code snippet instead

Reply:
hmm, i genuinely think that  showing all the examples would be clearer to integrators. What are the advantages we wanna combine all into a few?